services/mentor_service.js

```diff
@@ -73,8 +73,44 @@ async function getEligibleMentorStatuses(userRoles, isOwner) {
   }
 }
 
+async function updateMentorStatus(userRoles, mentorId, status, isOwner) {
+  const mentorData = await getMentorById(userRoles, mentorId, isOwner);
+  const isAdmin = userRoles.some(role => role.role_name === ROLE_NAMES.admin);
+  const allowedStatusesForOwner = [
+    MENTOR_STATUSES.active,
+    MENTOR_STATUSES.inactive,
+  ];
+
+  if(isAdmin) {
+    // if admin approves mentor (pending -> active -> add mentor role)
+    if(mentorData.status === MENTOR_STATUSES.pending && status === MENTOR_STATUSES.active) {
+      repo.updateMentorById(mentorId, { status });
+      assignRoleToUser(mentorData.user_id, 2);
+      return "Mentor application approved"
+    } else if(status === MENTOR_STATUSES.rejected) {
+      // if admin rejects mentor (change mentor status to rejected => remove mentor role)
+      repo.updateMentorById(mentorId, { status });
+      removeRoleFromUser(mentorData.user_id, 2);
```

Comment:
did you check what happens if we try to remove a role that doesn't exist? 
just please let's double check that we won't throw an exception here